terraform/modules/cluster/mig/variables.tf

```diff
@@ -219,33 +228,7 @@ variable "machine_image" {
     name    = string
     project = string
   })
-  default = {
-    project = "deeplearning-platform-release"
-    family  = "pytorch-latest-gpu-debian-11-py310"
-    name    = null
-  }
-
-  validation {
-    condition = (
-      var.machine_image != null
-      // project is non-empty
-      && alltrue([
-        for empty in [null, ""]
-        : var.machine_image.project != empty
-      ])
-      // at least one is non-empty
-      && anytrue([
-        for value in [var.machine_image.name, var.machine_image.family]
-        : alltrue([for empty in [null, ""] : value != empty])
-      ])
-      // at least one is empty
-      && anytrue([
-        for value in [var.machine_image.name, var.machine_image.family]
-        : anytrue([for empty in [null, ""] : value == empty])
-      ])
-    )
-    error_message = "project must be non-empty exactly one of family or name must be non-empty"
-  }
+  default = null
```

Comment:
Should we check for exclusivity ? If container_image is specified then machine_image should be null or machine_image.project = "cos-cloud"

Reply:
i was thinking we should leave that up to the user. if they want to have a container running on debian, then who am i to stop them?

Reply:
Agree, user might want to use some other image which is not COS and supports docker (say DLVM for that matter).
If they choose an OS that does not have docker and it fails, then it is a user error.

Reply:
As we discussed we should add a check If container_image is specified then machine_image should be null or machine_image.project = "cos-cloud". Since we dont want to add extra complexity of making sure containers work in other host OSes.